pkg/apis/serving/v1beta1/inference_service_validation.go

```diff
@@ -141,6 +147,57 @@ func validateInferenceService(isvc *InferenceService) (admission.Warnings, error
 	return allWarnings, nil
 }
 
+// validateMultiNodeVariables validates when there is workerSpec set in isvc
+func validateMultiNodeVariables(isvc *InferenceService) error {
+	if isvc.Spec.Predictor.WorkerSpec != nil {
+		if isvc.Spec.Predictor.Model != nil {
+			if _, exists := utils.GetEnvVarValue(isvc.Spec.Predictor.Model.PredictorExtensionSpec.Container.Env, constants.PipelineParallelSizeEnvName); exists {
+				return fmt.Errorf(DisallowedWorkerSpecPipelineParallelSizeEnvError, isvc.Name)
+			}
+			if _, exists := utils.GetEnvVarValue(isvc.Spec.Predictor.Model.PredictorExtensionSpec.Container.Env, constants.TensorParallelSizeEnvName); exists {
+				return fmt.Errorf(DisallowedWorkerSpecTensorParallelSizeEnvError, isvc.Name)
+			}
+			if utils.IsUnknownGpuResourceType(isvc.Spec.Predictor.Model.Resources) {
+				return fmt.Errorf(InvalidUnknownGPUTypeError, isvc.Name)
+			}
+			if isvc.Spec.Predictor.Model.StorageURI == nil {
+				return fmt.Errorf(MissingStorageURI, isvc.Name)
+			} else {
+				storageProtocol := strings.Split(*isvc.Spec.Predictor.Model.StorageURI, "://")[0]
+				if storageProtocol != "pvc" {
+					return fmt.Errorf(InvalidNotSupportedStorageURIProtocolError, isvc.Name, storageProtocol)
+				}
+			}
+			if isvc.GetAnnotations()[constants.AutoscalerClass] != string(constants.AutoscalerClassExternal) {
+				return fmt.Errorf(InvalidAutoScalerError, isvc.Name, isvc.GetAnnotations()[constants.AutoscalerClass])
+			}
+		}
+
+		// WorkerSpec.PipelineParallelSize should not be less than 2 (head + worker)
+		if isvc.Spec.Predictor.WorkerSpec.PipelineParallelSize != nil {
```

Comment:
I think `PipelineParallelSize` should be under `workerSpec` because it is related to multi-node setups. For `TensorParallelSize`, I’m okay with moving it.

The role of `TensorParallelSize` is to align the GPU resource count, which is particularly useful when there are two deployments, such as head/worker, ensuring the GPU resources are matched across both. In a single-node setup, it’s simpler because you only need to align resources within one deployment.

So, I believe both should remain under `workerSpec`. However, if we have to move one, I think it would be better to move only `TensorParallelSize`.

Reply:
if it needs to be changed, we can work with another pr.For now, I will resolve this conversation.